tests/Unit/Domain/CoordinateMaps/Test_UniformCylindricalFlatEndcap.cpp

```diff
@@ -0,0 +1,171 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <array>
+#include <cmath>
+#include <optional>
+#include <random>
+
+#include "Domain/CoordinateMaps/UniformCylindricalFlatEndcap.hpp"
+#include "Helpers/Domain/CoordinateMaps/TestMapHelpers.hpp"
+#include "NumericalAlgorithms/RootFinding/TOMS748.hpp"
+
+namespace domain {
+
+namespace {
+
+double maximum_rho_for_invertibility(double rho_max_so_far, double z_center_two,
+                                     const std::array<double, 3>& center_one,
+                                     double radius_one, double radius_two,
+                                     double theta_max_one) {
+  const auto is_invertible = [&](const double rho) {
+    // Direction shouldn't matter here, so just choose x-axis
+    return CoordinateMaps::
+                   is_uniform_cylindrical_flat_endcap_invertible_on_sphere_one(
+                       center_one,
+                       {{center_one[0] + rho, center_one[1], z_center_two}},
+                       radius_one, radius_two, theta_max_one)
+               ? 1.0
+               : -1.0;
+  };
+  const double invertible_at_zero = is_invertible(0.0);
+  const double invertible_at_max = is_invertible(rho_max_so_far);
+  ASSERT(invertible_at_zero == 1.0, "Map should always be invertible at rho=0");
```

Comment:
Why is this test an ASSERT instead of a CHECK? If you're running the test, don't you always want to check this?